udp/server.go

```diff
@@ -421,11 +421,22 @@ func (s *Server) getOrCreateClientConn(UDPConn *coapNet.UDPConn, raddr *net.UDPA
 }
 
 func (s *Server) getClientConn(l *coapNet.UDPConn, raddr *net.UDPAddr) *client.ClientConn {
+	if s.getListener() == nil {
+		// server is not started/stopped
+		return nil
+	}
 	cc, created := s.getOrCreateClientConn(l, raddr)
 	if created {
 		if s.onNewClientConn != nil {
 			s.onNewClientConn(cc)
 		}
+		return cc
+	}
+
+	if cc.Context().Err() != nil {
+		// connection is closed so we need to create new one
+		getClose(cc)()
+		return s.getClientConn(l, raddr)
```

Comment:
No chance of infinite recursion, I hope? :-)

Reply:
fixed 👍